### PR TITLE
feat(authentication-jwt) Add signup / create user

### PR DESCRIPTION
**Pull request by HrithikMittal**
This pull request is to provide the signup method in the authentication-jwt So it would be easier for developers.
https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by dhmlau** [comment]
Related to: https://github.com/strongloop/loopback-next/issues/5602

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 1]
```diff
@@ -45,8 +57,43 @@ export class UserController {
     public userService: UserService<User, Credentials>,
     @inject(SecurityBindings.USER, {optional: true})
     private user: UserProfile,
+    @inject(UserServiceBindings.USER_REPOSITORY)
+    public userRepository: UserRepository
   ) {}
 
+  @post('/signup', {
```
Comment (dougal83) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:64]: Personally I'd prefix with `users` resulting in `/users/signup` endpoint for clarity.  Not essential, to change.  Up to you.

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 2]
```diff
@@ -45,8 +57,43 @@ export class UserController {
...
+        'application/json': {
+          schema: CredentialsSchema,
+        },
+      },
+    })
+    newUserRequest: NewUserRequest,
+  ): Promise<User> {
+    const password = await hash(newUserRequest.password, await genSalt());
+    delete newUserRequest.password;
+    var savedUser = await this.userRepository.create(newUserRequest);
```
Comment (dougal83) [inline extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:90]:
```suggestion
    const savedUser = await this.userRepository.create(newUserRequest);
```

Comment (HrithikMittal) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts]: I make the changes you suggested please check now.

**Comment by dhmlau** [comment]
@HrithikMittal, thanks for the PR. Could you please take a look at the code linter and commit linter issue? 
For code linter, you can run `npm run lint:fix` to fix it.
For commit linter, please refer to https://loopback.io/doc/en/lb4/code-contrib-lb4.html#commit-message-guidelines. I'd suggest to add the `feat(authentication-jwt)` prefix to the commit message. 

Thanks!

**Comment by dhmlau** [comment]
@jannyHou, assigning this to you for the additional approval. Thanks.

File: extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts [hunk 3]
```diff
@@ -6,12 +6,15 @@
...
   TokenService,
-  UserService,
+  UserService
 } from '@loopback/authentication';
 import {inject} from '@loopback/core';
 import {get, post, requestBody} from '@loopback/rest';
 import {SecurityBindings, securityId, UserProfile} from '@loopback/security';
+import {genSalt, hash} from 'bcryptjs';
 import {TokenServiceBindings, User, UserServiceBindings} from '../../../';
+import {model, property} from '../../../../../../packages/repository/src';
```
[the remark this answers is not in the document]
Comment (HrithikMittal) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts:16]: Thanks @jannyHou I do the changes asap.

[the remark this answers is not in the document]
Comment (HrithikMittal) [reply extensions/authentication-jwt/src/__tests__/fixtures/controllers/user.controller.ts]: Please check it now @jannyHou

**Comment by jannyHou** [comment]
Hi @HrithikMittal any update on this PR? :) almost ready to land

**Comment by HrithikMittal** [comment]
> Hi @HrithikMittal any update on this PR? :) almost ready to land

Please if you can clean it. Please do it.

**Comment by jannyHou** [comment]
🎉 @HrithikMittal thank you! merged.

**Pull request merged**